static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,90 @@
+define([
+    'common/utils/fastdom-promise',
+    'Promise',
+    'common/utils/$',
+    'common/utils/mediator',
+    'common/utils/detect'
+], function (
+    fastdom,
+    Promise,
+    $,
+    mediator,
+    detect) {
+
+    return function () {
+        if (detect.getBreakpoint() !== 'mobile') {
+            fastdom.read(function () {
```

Comment:
Yes, so in your case

```
Promise.resolve([
  $('.js-top-banner-above-nav'),
  $('.js-header')
]).then(function (elems) {
  var $adBanner = elems[0];
  var $header = elems[1];
  return fastdom.read(function () {
    return [$adBanner.height(), $header.height()];
  });
}).then(...)
```

This is to reduce as much as possible the time needed to run the fastdom callback. It's a bit clearer I think since `fastdom.read` has been recently renamed `fastdom.measure`.